### ISSUE TITLE
[Paper Note] DeepSeek-V3 Technical Report, DeepSeek-AI+, arXiv'24, 2024.12

### ISSUE THREAD

**Issue by AkihikoWatanabe** [AkihikoWatanabe/paper_notes#1620]
# URL
- https://arxiv.org/abs/2412.19437
# Authors
  - DeepSeek-AI
  - Aixin Liu
  - Bei Feng
  - Bing Xue
  - Bingxuan Wang
  - Bochao Wu
  - Chengda Lu
  - Chenggang Zhao
  - Chengqi Deng
  - Chenyu Zhang
  - Chong Ruan
  - Damai Dai
  - Daya Guo
  - Dejian Yang
  - Deli Chen
  - Dongjie Ji
  - Erhang Li
  - Fangyun Lin
  - Fucong Dai
  - Fuli Luo
  - Guangbo Hao
  - Guanting Chen
  - Guowei Li
  - H. Zhang
  - Han Bao
  - Hanwei Xu
  - Haocheng Wang
  - Haowei Zhang
  - Honghui Ding
  - Huajian Xin
  - Huazuo Gao
  - Hui Li
  - Hui Qu
  - J. L. Cai
  - Jian Liang
  - Jianzhong Guo
  - Jiaqi Ni
  - Jiashi Li
  - Jiawei Wang
  - Jin Chen
  - Jingchang Chen
  - Jingyang Yuan
  - Junjie Qiu
  - Junlong Li
  - Junxiao Song
  - Kai Dong
  - Kai Hu
  - Kaige Gao
  - Kang Guan
  - Kexin Huang
  - Kuai Yu
  - Lean Wang
  - Lecong Zhang
  - Lei Xu
  - Leyi Xia
  - Liang Zhao
  - Litong Wang
  - Liyue Zhang
  - Meng Li
  - Miaojun Wang
  - Mingchuan Zhang
  - Minghua Zhang
  - Minghui Tang
  - Mingming Li
  - Ning Tian
  - Panpan Huang
  - Peiyi Wang
  - Peng Zhang
  - Qiancheng Wang
  - Qihao Zhu
  - Qinyu Chen
  - Qiushi Du
  - R. J. Chen
  - R. L. Jin
  - Ruiqi Ge
  - Ruisong Zhang
  - Ruizhe Pan
  - Runji Wang
  - Runxin Xu
  - Ruoyu Zhang
  - Ruyi Chen
  - S. S. Li
  - Shanghao Lu
  - Shangyan Zhou
  - Shanhuang Chen
  - Shaoqing Wu
  - Shengfeng Ye
  - Shengfeng Ye
  - Shirong Ma
  - Shiyu Wang
  - Shuang Zhou
  - Shuiping Yu
  - Shunfeng Zhou
  - Shuting Pan
  - T. Wang
  - Tao Yun
  - Tian Pei
  - Tianyu Sun
  - W. L. Xiao
  - Wangding Zeng
  - Wanjia Zhao
  - Wei An
  - Wen Liu
  - Wenfeng Liang
  - Wenjun Gao
  - Wenqin Yu
  - Wentao Zhang
  - X. Q. Li
  - Xiangyue Jin
  - Xianzu Wang
  - Xiao Bi
  - Xiaodong Liu
  - Xiaohan Wang
  - Xiaojin Shen
  - Xiaokang Chen
  - Xiaokang Zhang
  - Xiaosha Chen
  - Xiaotao Nie
  - Xiaowen Sun
  - Xiaoxiang Wang
  - Xin Cheng
  - Xin Liu
  - Xin Xie
  - Xingchao Liu
  - Xingkai Yu
  - Xinnan Song
  - Xinxia Shan
  - Xinyi Zhou
  - Xinyu Yang
  - Xinyuan Li
  - Xuecheng Su
  - Xuheng Lin
  - Y. K. Li
  - Y. Q. Wang
  - Y. X. Wei
  - Y. X. Zhu
  - Yang Zhang
  - Yanhong Xu
  - Yanhong Xu
  - Yanping Huang
  - Yao Li
  - Yao Zhao
  - Yaofeng Sun
  - Yaohui Li
  - Yaohui Wang
  - Yi Yu
  - Yi Zheng
  - Yichao Zhang
  - Yifan Shi
  - Yiliang Xiong
  - Ying He
  - Ying Tang
  - Yishi Piao
  - Yisong Wang
  - Yixuan Tan
  - Yiyang Ma
  - Yiyuan Liu
  - Yongqiang Guo
  - Yu Wu
  - Yuan Ou
  - Yuchen Zhu
  - Yuduan Wang
  - Yue Gong
  - Yuheng Zou
  - Yujia He
  - Yukun Zha
  - Yunfan Xiong
  - Yunxian Ma
  - Yuting Yan
  - Yuxiang Luo
  - Yuxiang You
  - Yuxuan Liu
  - Yuyang Zhou
  - Z. F. Wu
  - Z. Z. Ren
  - Zehui Ren
  - Zhangli Sha
  - Zhe Fu
  - Zhean Xu
  - Zhen Huang
  - Zhen Zhang
  - Zhenda Xie
  - Zhengyan Zhang
  - Zhewen Hao
  - Zhibin Gou
  - Zhicheng Ma
  - Zhigang Yan
  - Zhihong Shao
  - Zhipeng Xu
  - Zhiyu Wu
  - Zhongyu Zhang
  - Zhuoshu Li
  - Zihui Gu
  - Zijia Zhu
  - Zijun Liu
  - Zilin Li
  - Ziwei Xie
  - Ziyang Song
  - Ziyi Gao
  - Zizheng Pan
# Abstract
  - We present DeepSeek-V3, a strong Mixture-of-Experts (MoE) language model with 671B total parameters with 37B activated for each token. To achieve efficient inference and cost-effective training, DeepSeek-V3 adopts Multi-head Latent Attention (MLA) and DeepSeekMoE architectures, which were thoroughly validated in DeepSeek-V2. Furthermore, DeepSeek-V3 pioneers an auxiliary-loss-free strategy for load balancing and sets a multi-token prediction training objective for stronger performance. We pre-train DeepSeek-V3 on 14.8 trillion diverse and high-quality tokens, followed by Supervised Fine-Tuning and Reinforcement Learning stages to fully harness its capabilities. Comprehensive evaluations reveal that DeepSeek-V3 outperforms other open-source models and achieves performance comparable to leading closed-source models. Despite its excellent performance, DeepSeek-V3 requires only 2.788M H800 GPU hours for its full training. In addition, its training process is remarkably stable. Throughout the entire training process, we did not experience any irrecoverable loss spikes or perform any rollbacks. The model checkpoints are available at https://github.com/deepseek-ai/DeepSeek-V3.
# Translation (by gpt-5-nano)
- DeepSeek-V3を紹介します。671Bの総パラメータを持ち、各トークンに対して37Bが活性化される強力なMixture-of-Experts (MoE)言語モデルです。推論を効率化し、訓練コストを抑えるため、DeepSeek-V3はMulti-head Latent Attention (MLA)およびDeepSeekMoEアーキテクチャを採用しており、これらはDeepSeek-V2で徹底的に検証されています。さらに、DeepSeek-V3はロードバランシングのための補助損失を用いない戦略を提案し、多トークン予測訓練目的を設定してより強力な性能を発揮します。DeepSeek-V3を14.8兆トークンの多様で高品質なデータでプレトレーニングした後、教師ありファインチューニングおよび強化学習段階を経て、その能力を最大限に活かします。総合的な評価により、DeepSeek-V3は他のオープンソースモデルを上回り、先端的なクローズドソースモデルと同等の性能を達成することが示されています。優れた性能にもかかわらず、DeepSeek-V3の全訓練にはわずか2,788,000時間分のH800 GPU時間が必要です。さらに、訓練プロセスは非常に安定しており、訓練全体を通じて回復不能な損失の急上昇やロールバックは発生しませんでした。モデルのチェックポイントはhttps://github.com/deepseek-ai/DeepSeek-V3で入手可能です。
# Summary (by gpt-4o-mini)
- DeepSeek-V3は671Bのパラメータを持つMixture-of-Experts (MoE)言語モデルで、各トークンに対して37Bが活性化される。効率的な推論とコスト削減のため、MLAおよびDeepSeekMoEアーキテクチャを採用し、補助損失を用いない戦略を導入。14.8兆トークンでプレトレーニング後、ファインチューニングと強化学習を経て、高性能を発揮。評価結果はオープンソースモデルを上回り、先端的なクローズドソースモデルとも同等。訓練にはわずか2,788,000時間のH800 GPU時間を要し、安定した訓練プロセスを実現。モデルのチェックポイントは提供されている。

**Comment by AkihikoWatanabe** [comment]
参考（モデルの図解）:https://x.com/vtabbott_/status/1874449446056177717?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
参考:https://x.com/hillbig/status/1876397959841186148?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q

**Comment by AkihikoWatanabe** [comment]
MLA(Multi-Head Latent Attention)を提案

解説:
- #1621
- #1663

MLAはKVを低ランクなlatentベクトルに圧縮して保持し、使う時に復元するといった操作をすることで、MHAのパフォーマンスを落とすことなくKV Cacheで利用するメモリを大幅に減らせるという手法。

**Comment by AkihikoWatanabe** [comment]
MLAの図解:
https://x.com/eliebakouch/status/2032146159104405851?s=46&t=Y6UuIHB0Lv0IpmFAjlc2-Q
